### PR TITLE
strtok, strtok_r, and strsep have restrict arrguments

### PR DESCRIPTION
**Pull request by AtariDreams**
This PR also includes expect_strtok for good measure since it is just strtok but modified to skip \ sequences.

### CODE REVIEW

**Comment by rilysh** [comment]
Out of curiosity, but where did you find that these functions requires `__restrict`?

Man pages don't seem to inform that though. I assume compilers can do such optimizations automatically (in most cases).

strtok(3): https://man.freebsd.org/cgi/man.cgi?query=strtok&apropos=0&sektion=3&manpath=FreeBSD+15.0-CURRENT&arch=default&format=html

strsep(3): https://man.freebsd.org/cgi/man.cgi?query=strsep&apropos=0&sektion=3&manpath=FreeBSD+15.0-CURRENT&arch=default&format=html

File: usr.bin/chat/chat.c
Comment (igalic) [inline usr.bin/chat/chat.c:1]: this change is unrelated, and should be removed

**Review by igalic — COMMENTED** [review]
i don't know if this will need a version bump, or an exp-run to ensure we're not accidentally (un)breaking anything

**Comment by AtariDreams** [comment]
> Out of curiosity, but where did you find that these functions requires `__restrict`?
> 
> Man pages don't seem to inform that though. I assume compilers can do such optimizations automatically (in most cases).
> 
> strtok(3): https://man.freebsd.org/cgi/man.cgi?query=strtok&apropos=0&sektion=3&manpath=FreeBSD+15.0-CURRENT&arch=default&format=html
> 
> strsep(3): https://man.freebsd.org/cgi/man.cgi?query=strsep&apropos=0&sektion=3&manpath=FreeBSD+15.0-CURRENT&arch=default&format=html

Source: https://man7.org/linux/man-pages/man3/strsep.3.html

File: usr.bin/chat/chat.c
Comment (AtariDreams) [reply usr.bin/chat/chat.c:1]: "A modified version of 'strtok'. This version skips \ sequences."

I think it should count.

**Comment by rilysh** [comment]
> > Out of curiosity, but where did you find that these functions requires `__restrict`?
> > Man pages don't seem to inform that though. I assume compilers can do such optimizations automatically (in most cases).
> > strtok(3): https://man.freebsd.org/cgi/man.cgi?query=strtok&apropos=0&sektion=3&manpath=FreeBSD+15.0-CURRENT&arch=default&format=html
> > strsep(3): https://man.freebsd.org/cgi/man.cgi?query=strsep&apropos=0&sektion=3&manpath=FreeBSD+15.0-CURRENT&arch=default&format=html
> 
> Source: https://man7.org/linux/man-pages/man3/strsep.3.html

That's Linux man page, strictly glibc focused. Other libc's such as musl don't force `__restrict` unnecessary (including in strsep()).

**Comment by AtariDreams** [comment]
> > > Out of curiosity, but where did you find that these functions requires `__restrict`?
> 
> > > Man pages don't seem to inform that though. I assume compilers can do such optimizations automatically (in most cases).
> 
> > > strtok(3): https://man.freebsd.org/cgi/man.cgi?query=strtok&apropos=0&sektion=3&manpath=FreeBSD+15.0-CURRENT&arch=default&format=html
> 
> > > strsep(3): https://man.freebsd.org/cgi/man.cgi?query=strsep&apropos=0&sektion=3&manpath=FreeBSD+15.0-CURRENT&arch=default&format=html
> 
> > 
> 
> > Source: https://man7.org/linux/man-pages/man3/strsep.3.html
> 
> 
> 
> That's Linux man page, strictly glibc focused. Other libc's such as musl don't force `__restrict` unnecessary (including in strsep()).

Oracle says it too for Solaris:

https://docs.oracle.com/cd/E88353_01/html/E37843/strsep-3c.html

**Comment by bsdimp** [comment]
So this change is completely unacceptable in its current form.
There's way too many changes merged together, that makes it impossible to review.
Please do things one at a time, one commit per change.
But even one commit per change, the theme pulling these together is only that they deal with restrict, which may be a bit over-broad.
Please resubmit a pull request for strtok_r, and another for printf, etc. I'm not sure that all the strtoX ones are even correct. For a change like this, you should likely not be changing more than 3 or 4 files per pull request. 
Some of this might be good, but the breath of this change makes it hard to find, and increases the risk of something slipping through that shouldn't.
